### DISCUSSION TITLE
Error Exit: Unable to open SU2 restart file restart_adj_cd.dat.

### DISCUSSION THREAD

**Discussion by 1Northward1** [su2code/SU2#1774] [Q&A]
**Hi, 

I am trying to optimize an airfoil and the Mach number is 0.1. Therefore, the INC_RANS solver was chosen.
The simulation results are excellent, but the error occurs when I optimize it:

The output of the terminal is as follows:**

SU2.EvaluationFailure: Path = /home/Mars/SU2/Flatback_incompressible/Opt_test2/DESIGNS/DSN_001/ADJOINT_DRAG/,
Command = mpirun -n 40 /home/Mars/SU2/SU2-v7.4.0-linux64-mpi/bin/SU2_SOL config_SOL.cfg
SU2 process returned error '1'

**If choosing compressible RANS as the Solver, the above problems will not occur. But the Mach number is too small (0.1), and it is difficult to converge when using the compressible RANS solver.**

**The messages in the DESIGNS/DSN_001/ADJOINT_DRAG/log_Adjoint.out are as follows:**
.
.
.
-------------------------------------------------------------------------
------------------------- Exit Success (SU2_CFD) ------------------------
.
.
.
------------------------- Solution Postprocessing -----------------------

Error in "void CBaselineSolver::SetOutputVariables(CGeometry*, CConfig*)": 
-------------------------------------------------------------------------
Unable to open SU2 restart file restart_adj_cd.dat
------------------------------ Error Exit -------------------------------

**My configuration file is as follows:**

% ------------- DIRECT, ADJOINT, AND LINEARIZED PROBLEM DEFINITION ------------%
SOLVER= INC_RANS
KIND_TURB_MODEL= SA
MATH_PROBLEM= DIRECT
RESTART_SOL= YES
% -------------------- INCOMPRESSIBLE FREE-STREAM DEFINITION ------------------%
INC_DENSITY_INIT= 1.5781
INC_VELOCITY_INIT= ( 34.03, 0.0, 0.0 )
INC_NONDIM= INITIAL_VALUES
INC_DENSITY_REF= 1.0
INC_VELOCITY_REF= 1.0
% --------------------------- FLUID PROPERTIES ---------------------------------%
FLUID_MODEL= CONSTANT_DENSITY
VISCOSITY_MODEL= CONSTANT_VISCOSITY
MU_CONSTANT= 1.7894e-05
% ---------------------- REFERENCE VALUE DEFINITION ---------------------------%
REF_ORIGIN_MOMENT_X = 0.05
REF_ORIGIN_MOMENT_Y = 0.00
REF_ORIGIN_MOMENT_Z = 0.00
REF_LENGTH= 1.0
REF_AREA= 1.0
% -------------------- BOUNDARY CONDITION DEFINITION --------------------------%
MARKER_HEATFLUX= ( airfoil, 0.0 )
MARKER_FAR= ( farfield )
MARKER_PLOTTING= ( airfoil )
MARKER_MONITORING= ( airfoil )
% ------------- COMMON PARAMETERS DEFINING THE NUMERICAL METHOD ---------------%
NUM_METHOD_GRAD= GREEN_GAUSS
CFL_ADAPT= YES
CFL_ADAPT_PARAM= ( 0.1, 1.2, 50.0, 200 )
MAX_DELTA_TIME= 1E6
RK_ALPHA_COEFF= ( 0.66667, 0.66667, 1.000000 )
ITER= 5000
% ------------------------ LINEAR SOLVER DEFINITION ---------------------------%
LINEAR_SOLVER= FGMRES
LINEAR_SOLVER_PREC= ILU
LINEAR_SOLVER_ILU_FILL_IN= 0
LINEAR_SOLVER_ERROR= 1E-10
LINEAR_SOLVER_ITER= 10
% -------------------------- MULTIGRID PARAMETERS -----------------------------%
MGLEVEL= 3
MGCYCLE= W_CYCLE
MG_PRE_SMOOTH= ( 1, 2, 3, 3 )
MG_POST_SMOOTH= ( 0, 0, 0, 0 )
MG_CORRECTION_SMOOTH= ( 0, 0, 0, 0 )
MG_DAMP_RESTRICTION= 0.8
MG_DAMP_PROLONGATION= 0.8
% -------------------- FLOW NUMERICAL METHOD DEFINITION -----------------------%
CONV_NUM_METHOD_FLOW= JST
SLOPE_LIMITER_FLOW= VENKATAKRISHNAN
JST_SENSOR_COEFF= ( 0.5, 0.02 )
TIME_DISCRE_FLOW= EULER_IMPLICIT
% -------------------- TURBULENT NUMERICAL METHOD DEFINITION ------------------%
CONV_NUM_METHOD_TURB= SCALAR_UPWIND
MUSCL_TURB= NO
SLOPE_LIMITER_TURB= VENKATAKRISHNAN
TIME_DISCRE_TURB= EULER_IMPLICIT
% ---------------- ADJOINT-FLOW NUMERICAL METHOD DEFINITION -------------------%
OBJECTIVE_FUNCTION= DRAG
CONV_NUM_METHOD_ADJFLOW= JST
SLOPE_LIMITER_ADJFLOW= VENKATAKRISHNAN
ADJ_JST_SENSOR_COEFF= ( 0.0, 0.02 )
TIME_DISCRE_ADJFLOW= EULER_IMPLICIT
CFL_REDUCTION_ADJFLOW= 0.8
LIMIT_ADJFLOW= 1E6
% ----------------------- GEOMETRY EVALUATION PARAMETERS ----------------------%
GEO_MARKER= ( airfoil )
GEO_DESCRIPTION= AIRFOIL
GEO_MODE= FUNCTION
% ----------------------- DESIGN VARIABLE PARAMETERS --------------------------%
DV_KIND= HICKS_HENNE
DV_MARKER= ( airfoil )
DV_PARAM= ( 1, 0.5 )
DV_VALUE= 1.0
% ------------------------ GRID DEFORMATION PARAMETERS ------------------------%
DEFORM_LINEAR_SOLVER_ITER= 500
DEFORM_NONLINEAR_ITER= 1
DEFORM_CONSOLE_OUTPUT= YES
DEFORM_LINEAR_SOLVER_ERROR= 1E-14
DEFORM_STIFFNESS_TYPE= INVERSE_VOLUME
% --------------------------- CONVERGENCE PARAMETERS --------------------------%
CONV_FIELD= LIFT, DRAG, MOMENT_Z
CONV_RESIDUAL_MINVAL= -10
CONV_STARTITER= 500
CONV_CAUCHY_ELEMS= 100
CONV_CAUCHY_EPS= 1E-6
% ------------------------- INPUT/OUTPUT INFORMATION --------------------------%
MESH_FILENAME= DU97-Flatback.su2
MESH_FORMAT= SU2
MESH_OUT_FILENAME= mesh_out.su2
SOLUTION_FILENAME= solution_flow.dat
SOLUTION_ADJ_FILENAME= solution_adj.dat
TABULAR_FORMAT= CSV
CONV_FILENAME= history
RESTART_FILENAME= restart_flow.dat
RESTART_ADJ_FILENAME= restart_adj.dat
VOLUME_FILENAME= flow
VOLUME_ADJ_FILENAME= adjoint
GRAD_OBJFUNC_FILENAME= of_grad.dat
SURFACE_FILENAME= surface_flow
SURFACE_ADJ_FILENAME= surface_adjoint
SCREEN_OUTPUT = (INNER_ITER, RMS_PRESSURE, LIFT, DRAG, MOMENT_Z)
OUTPUT_FILES = (RESTART, PARAVIEW, SURFACE_PARAVIEW, SURFACE_CSV)
% --------------------- OPTIMAL SHAPE DESIGN DEFINITION -----------------------%
OPT_OBJECTIVE= DRAG
OPT_CONSTRAINT= NONE
OPT_GRADIENT_FACTOR= 1E-6
OPT_RELAX_FACTOR= 1E3
OPT_ITERATIONS= 100
OPT_ACCURACY= 1E-10
OPT_BOUND_UPPER= 0.1
OPT_BOUND_LOWER= -0.1
DEFINITION_DV= ( 30, 1.0 | airfoil | 0, 0.05 ); ( 30, 1.0 | airfoil | 0, 0.10 ); ( 30, 1.0 | airfoil | 0, 0.15 ); ( 30, 1.0 | airfoil | 0, 0.20 ); ( 30, 1.0 | airfoil | 0, 0.25 ); ( 30, 1.0 | airfoil | 0, 0.30 ); ( 30, 1.0 | airfoil | 0, 0.35 ); ( 30, 1.0 | airfoil | 0, 0.40 ); ( 30, 1.0 | airfoil | 0, 0.45 ); ( 30, 1.0 | airfoil | 0, 0.50 ); ( 30, 1.0 | airfoil | 0, 0.55 ); ( 30, 1.0 | airfoil | 0, 0.60 ); ( 30, 1.0 | airfoil | 0, 0.65 ); ( 30, 1.0 | airfoil | 0, 0.70 ); ( 30, 1.0 | airfoil | 0, 0.75 ); ( 30, 1.0 | airfoil | 0, 0.80 ); ( 30, 1.0 | airfoil | 0, 0.85 ); ( 30, 1.0 | airfoil | 0, 0.90 ); ( 30, 1.0 | airfoil | 0, 0.95 ); ( 30, 1.0 | airfoil | 1, 0.05 ); ( 30, 1.0 | airfoil | 1, 0.10 ); ( 30, 1.0 | airfoil | 1, 0.15 ); ( 30, 1.0 | airfoil | 1, 0.20 ); ( 30, 1.0 | airfoil | 1, 0.25 ); ( 30, 1.0 | airfoil | 1, 0.30 ); ( 30, 1.0 | airfoil | 1, 0.35 ); ( 30, 1.0 | airfoil | 1, 0.40 ); ( 30, 1.0 | airfoil | 1, 0.45 ); ( 30, 1.0 | airfoil | 1, 0.50 ); ( 30, 1.0 | airfoil | 1, 0.55 ); ( 30, 1.0 | airfoil | 1, 0.60 ); ( 30, 1.0 | airfoil | 1, 0.65 ); ( 30, 1.0 | airfoil | 1, 0.70 ); ( 30, 1.0 | airfoil | 1, 0.75 ); ( 30, 1.0 | airfoil | 1, 0.80 ); ( 30, 1.0 | airfoil | 1, 0.85 ); ( 30, 1.0 | airfoil | 1, 0.90 ); ( 30, 1.0 | airfoil | 1, 0.95 )

**I tried to solve this problem but failed.
 I would appreciate it if you help me.

Best regards,
Mars**

**Comment by bigfooted** [comment]
do you have a restart file? If so, what it the name of that restart file? And in the case of compressible flow, is the restart filename different?

**Reply by 1Northward1** [reply]
Thank you for your help! 
I just followed the tutorial. The tutorial does not mention the restart file. I was confused.
I plan to try again after compiling SU2 from the source code.
Anyway, thank you very much for your answer！

**Reply by xiapxiaoyu1981** [reply]
dear bigfooted , 1Northward1
I have the same question. 
After completing the CFD analysis and obtaining the restart_flow.dat file, the main program was interrupted. Upon inspecting the log_Direct.out file, it showed (at the end.
) ' ------------------------- Solution Postprocessing -----------------------
Reading and storing the solution from restart_flow.


Error in "virtual void CBaselineOutput::SetVolumeOutputFields(CConfig*)": 
-------------------------------------------------------------------------
No coordinates found in the restart file!!
------------------------------ Error Exit -------------------------------


Identify vertices.
Storing a mapping from global to local point index.

------------------------- Solution Postprocessing -----------------------
Reading and storing the solution from restart_flow.


Error in "virtual void CBaselineOutput::SetVolumeOutputFields(CConfig*)": 
-------------------------------------------------------------------------
No coordinates found in the restart file!!
------------------------------ Error Exit -------------------------------


Identify vertices.
Storing a mapping from global to local point index.

------------------------- Solution Postprocessing -----------------------
Reading and storing the solution from restart_flow.


Error in "virtual void CBaselineOutput::SetVolumeOutputFields(CConfig*)": 
-------------------------------------------------------------------------
No coordinates found in the restart file!!
------------------------------ Error Exit -------------------------------”' 
Upon entering the relevant directory and executing 'mpirun -n 4 /home/tongtong/su2/SU2CODE/bin/SU2_SOL config_SOL.cfg', it will display ““------------------------- Solution Postprocessing -----------------------
Reading and storing the solution from restart_flow.
Identify vertices.


Error in "virtual void CBaselineOutput::SetVolumeOutputFields(CConfig*)":
-------------------------------------------------------------------------
No coordinates found in the restart file!!
------------------------------ Error Exit -------------------------------


Storing a mapping from global to local point index.

------------------------- Solution Postprocessing -----------------------
Reading and storing the solution from restart_flow.
--------------------------------------------------------------------------
Primary job terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------


Error in "virtual void CBaselineOutput::SetVolumeOutputFields(CConfig*)":
-------------------------------------------------------------------------
No coordinates found in the restart file!!
------------------------------ Error Exit -------------------------------


--------------------------------------------------------------------------
mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

 Process name: [[64059,1],0]”

"Executing the command 'mpirun -n 4 /home/tongtong/su2/SU2CODE/bin/SU2_CFD config_SOL.cfg' enables the computation to proceed." (Whether SU2_CFD is more versatile than SU2_SOL depends on the specific context.)
